presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcRecordSetProvider.java

```diff
@@ -45,6 +47,13 @@ public RecordSet getRecordSet(ConnectorTransactionHandle transaction, ConnectorS
         JdbcSplit jdbcSplit = (JdbcSplit) split;
         JdbcTableHandle jdbcTable = (JdbcTableHandle) table;
 
+        // In the current API, the columns (and order) needed by the engine are provided via an argument to this method. Make sure that
+        // any columns that were recorded in the table handle match the requested set.
+        // If no columns are recorded, it means that applyProjection never got called (e.g., in the case all columns are being used) and all
+        // table columns should be returned. TODO: this is something that should be addressed once the getRecordSet API is revamped
+        jdbcTable.getColumns()
+                .ifPresent(tableColumns -> verify(ImmutableSet.copyOf(columns).equals(ImmutableSet.copyOf(tableColumns))));
```

Comment:
what about the order?

Reply:
Yes, good catch. I didn't recall if we made any guarantees, but the LocalExecutionPlanner respects the order of the output symbols of the TableScan. I'll update to check that, too.